crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -139,25 +147,29 @@ async fn create(
             )?;
         }
     }
-
-    let upsert = diesel::insert_into(default_configs)
-        .values(&default_config)
-        .on_conflict(db::schema::default_configs::key)
-        .do_update()
-        .set(&default_config)
-        .execute(&mut conn);
-
-    match upsert {
-        Ok(_) => Ok(HttpResponse::Ok().json(json!({
-            "message": "DefaultConfig created/updated successfully."
-        }))),
-        Err(e) => {
-            log::info!("DefaultConfig creation failed with error: {e}");
-            Err(unexpected_error!(
-                "Something went wrong, failed to create DefaultConfig"
-            ))
-        }
-    }
+    let version_id = generate_snowflake_id(&state)?;
```

Comment:
Can we generate version_id inside of add_config_version instead of repeating this code in each handler?